entities/storobj/storage_object.go

```diff
@@ -113,94 +111,81 @@ func FromBinaryUUIDOnly(data []byte) (*Object, error) {
+	vectorLength2 := rw.ReadUint16()
+	// The vector length should always be returned (for usage metrics purposes) even if the vector itself is skipped
+	ko.VectorLen = int(vectorLength2)
 	if addProp.Vector {
-		ko.Vector = make([]float32, vectorLength)
-		ec.AddWrap(binary.Read(r, le, &ko.Vector), "read vector")
-	} else {
-		io.CopyN(io.Discard, r, int64(vectorLength)*4)
-	}
-	ec.AddWrap(binary.Read(r, le, &classNameLength), "class name length")
-	className := make([]byte, classNameLength)
-	_, err = r.Read(className)
-	ec.AddWrap(err, "class name")
-	ec.AddWrap(binary.Read(r, le, &schemaLength), "schema length")
-	schema := make([]byte, schemaLength)
-	_, err = r.Read(schema)
-	ec.AddWrap(err, "schema")
-	ec.AddWrap(binary.Read(r, le, &metaLength), "additional length")
-	var meta []byte
-	if addProp.Classification || len(addProp.ModuleParams) > 0 {
-		meta = make([]byte, metaLength)
-		_, err = r.Read(meta)
-		ec.AddWrap(err, "read additional")
+		ko.Object.Vector = make([]float32, vectorLength2)
+		vectorBytes := rw.ReadBytesFromBuffer(uint64(vectorLength2) * 4)
+		for i := 0; i < int(vectorLength2); i++ {
+			bits := binary.LittleEndian.Uint32(vectorBytes[i*4 : (i+1)*4])
+			ko.Object.Vector[i] = math.Float32frombits(bits)
+		}
 	} else {
-		io.CopyN(io.Discard, r, int64(metaLength))
+		rw.MoveBufferPositionForward(uint64(vectorLength2) * 4)
+		ko.Object.Vector = nil
 	}
+	ko.Vector = ko.Object.Vector
 
-	ec.AddWrap(binary.Read(r, le, &vectorWeightsLength), "vector weights length")
-	vectorWeights := make([]byte, vectorWeightsLength)
-	_, err = r.Read(vectorWeights)
-	ec.AddWrap(err, "vector weights")
-
-	if err := ec.ToError(); err != nil {
-		return nil, errors.Wrap(err, "compound err")
-	}
+	classNameLen := rw.ReadUint16()
+	className := string(rw.ReadBytesFromBuffer(uint64(classNameLen)))
 
-	uuidParsed, err := uuid.FromBytes(uuidBytes)
-	if err != nil {
-		return nil, err
+	propLength := rw.ReadUint32()
+	var props []byte
+	if addProp.NoProps {
+		rw.MoveBufferPositionForward(uint64(propLength))
+	} else {
+		props = rw.ReadBytesFromBuffer(uint64(propLength))
 	}
 
-	if err := ko.parseObject(
-		strfmt.UUID(uuidParsed.String()),
-		createTime,
-		updateTime,
-		string(className),
-		schema,
-		meta,
-		vectorWeights,
-	); err != nil {
-		return nil, errors.Wrap(err, "parse")
+	var meta []byte
+	metaLength := rw.ReadUint32()
+	if addProp.Classification || len(addProp.ModuleParams) > 0 {
+		meta = rw.ReadBytesFromBuffer(uint64(metaLength))
+	} else {
+		rw.MoveBufferPositionForward(uint64(metaLength))
+	}
+
+	vectorWeightsLength := rw.ReadUint32()
+	vectorWeights := rw.ReadBytesFromBuffer(uint64(vectorWeightsLength))
+
+	// some object members need additional "enrichment". Only do this if necessary
+	if len(props) > 0 ||
+		len(meta) > 0 ||
+		vectorWeightsLength > 0 && !(vectorWeightsLength == 4 && vectorWeights[0] == 110 && vectorWeights[1] == 117 && vectorWeights[2] == 108 && vectorWeights[3] == 108) {
```

Comment:
Would help readability to break this down into separate conditional functions. This way you can use the naming of the functions to self-document the reasoning behind each of these propositions

Reply:
I reformated it and added comments to make it clearer